pkg/config/configuration.go

```diff
@@ -582,6 +595,18 @@ func (c Configuration) GetAPILoggingSpec() APILoggingSpec {
 	return *c.Spec.LoggingSpec.APILogging
 }
 
+// GetWorkflowSpec returns the Workflow spec.
+// It's a short-hand that includes nil-checks for safety.
+func (c *Configuration) GetWorkflowSpec() WorkflowSpec {
+	if c == nil || c.Spec.WorkflowSpec == nil {
+		return WorkflowSpec{
```

Comment:
I think default values should be handled by the consumer and not here.

The problem with this is that if you pass a configuration like:

```yaml
spec:
  workflow:
```

Or:

```yaml
spec:
  workflow:
    maxConcurrentWorkflowInvocations: 100
```

Or:

```yaml
spec:
  workflow:
    maxConcurrentActivityInvocations: 100
```

In all of the cases above, `c.Spec.WorkflowSpec` will not be nil, so _at least one of_ `MaxConcurrentWorkflowInvocations` and `MaxConcurrentActivityInvocations` will have a value of 0, because this `if` will not be invoked

Reply:
If you do want to keep the defaults in this file, perhaps consider creating a getter for each one of those properties.